samples/csharp_dotnetcore/06.using-cards/Bots/RichCardsBot.cs

```diff
@@ -1,39 +1,32 @@
 ﻿// Copyright (c) Microsoft Corporation. All rights reserved.
 // Licensed under the MIT License.
 
-using System.Collections.Generic;
 using System.Threading;
 using System.Threading.Tasks;
 using Microsoft.Bot.Builder;
-using Microsoft.Bot.Schema;
+using Microsoft.Bot.Builder.Dialogs;
 using Microsoft.Extensions.Logging;
 
-namespace Microsoft.BotBuilderSamples
+namespace Microsoft.BotBuilderSamples.Bots
 {
-    // RichCardsBot prompts a user to select a Rich Card and then returns the card
-    // that matches the user's selection.
-    public class RichCardsBot : DialogBot<MainDialog>
+    public class RichCardsBot<T> : ActivityHandler
```

Comment:
I think we want to move away from ActivityHandler, while using DialogManager ... since OnTurn is overridden, the other activity handler events will not fire.

imo we should just implement IBot while using DialogManager, unless there is a specific need to pre-process something within a specific event handler, before passing off the activity processing to DialogManager.

```cs
public class DialogBot<T> : IBot
        where T : Dialog
    {
        private readonly DialogManager DialogManager;
        protected readonly ILogger Logger;

        public DialogBot(T rootDialog, ILogger<DialogBot<T>> logger)
        {
            Logger = logger;

            DialogManager = new DialogManager(rootDialog);
        }

        public async Task OnTurnAsync(ITurnContext turnContext, CancellationToken cancellationToken = default)
        {
            Logger.LogInformation("Running dialog with Activity.");
            
            await DialogManager.OnTurnAsync(turnContext, cancellationToken: cancellationToken).ConfigureAwait(false);
        }
    }
``` 

https://github.com/microsoft/BotBuilder-Samples/blob/R11Samples/samples/csharp_dotnetcore/05.multi-turn-prompt/Bots/DialogBot.cs

Reply:
Changed. Thanks for the suggestion.